### PR TITLE
WIP: react sidebar

### PR DESCRIPTION
**Pull request by crptm**

### CODE REVIEW

File: app/scripts/react/balance/TokenBalances.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+var angularUtils = require("../angular-utils");
+var di = angularUtils.di;
+var TokenRow = require("./TokenRow");
+var AddCustomTokenForm = require("./AddCustomTokenForm");
+
+class TokenBalances extends React.Component {
+  constructor() {
```
Comment (dternyak) [inline app/scripts/react/balance/TokenBalances.js]: No PropTypes specified.

File: app/scripts/react/angular-utils.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  function($filter, $rootScope, $translate) {
+    translateService = $translate;
+    rootScope = $rootScope;
+    filters.number = $filter("number");
+    filters.currency = $filter("currency");
+    module.exports.di.rootScope = $rootScope;
+  }
+]);
+
+function _findInScope(scope, key) {
```
Comment (dternyak) [inline app/scripts/react/angular-utils.js]: Is there perhaps a battle-tested library for this? Ideally we would use a standardized approach for integrating React into Angular.

Comment (dternyak) [inline app/scripts/react/angular-utils.js:20]: Is there perhaps a battle-tested library for this? Ideally we would use a standardized approach for integrating React into Angular.  

e.g. https://github.com/ngReact/ngReact

We should do a thorough analysis on our options before me merge this, as there may be subtle bugs introduced without thoroughly vetting the integration.

File: app/scripts/react/balance/TokenBalances.js (hunk 1 above)
Comment (dternyak) [inline app/scripts/react/balance/TokenBalances.js]: No PropTypes specified.

**Review by dternyak — CHANGES_REQUESTED** [review]

File: app/scripts/react/angular-utils.js (hunk 2 above)
Comment (crptm) [reply app/scripts/react/angular-utils.js:20]: Yeah, so i started with ngReact and it is broken. We are using react2angular.

Regarding the issue - i think it is fine. Keep in mind we only need this for couple of months max.

Technically we can pass everything down as props, i'll try that for your peace of mind :)

**Pull request closed without merging**